File: android/app/src/main/java/io/ionic/starter/data/models/ApiResponse.kt
package io.ionic.starter.data.models

import android.util.Log
import java.io.Serializable

class ApiResponse : Serializable
{
    // Getters and setters
    var isSuccess: Boolean? = false
    var data: Data? = null




    class Data : Serializable
    {
        // Getters and setters
        var version: String? = null
        var user: String? = null
        var dateGenerated: String? = null
        var status: String? = null
        var data: List<WeatherData>? = null
    }

    class WeatherData : Serializable
    {
        // Getters and setters
        var lat: Double? = 0.0
        var lon: Double? = 0.0
        var date: String? = null
        var parameters: List<Parameter>? = null
        var coordinates: List<Coordinates?>? = null
        var parameter: String? = null
    }

    class Parameter : Serializable
    {
        // Getters and setters
        var parameter: String? = null
        var value: Double? = 0.0

    }

     class Coordinates:Serializable
     {
         var lat: Double ?  = null
         val lon: Double?  = null
         val dates: List<DateValue> = emptyList()
     }


}

File: ios/App/App/data/models/ApiResponse.swift
//
//  ApiResponse.swift
//  App
//
//  Created by Shahzaib Ali on 01/08/2024.
//

import Foundation

struct ApiResponse: Codable {
    var success: Bool? = nil
    var data: WeatherDataContainer? = nil
}

struct WeatherDataContainer: Codable {
    var version: String? = nil
    var user: String? = nil
    var dateGenerated: String? = nil
    var status: String? = nil
    var data: [WeatherParameter]? = nil
}

struct WeatherParameter: Codable {
    
    var parameter: String? = nil
    var parameters: [Parameter]? = nil
    var value: Double? = 0.0
    var coordinates: [WeatherCoordinate]? = nil
    
    
}

struct Parameter: Codable {
    var parameter: String? = nil
    var value: Double? = 0.0
}

struct WeatherCoordinate: Codable {
    var lat: Double? = nil
    var lon: Double? = nil
    var dates: [WeatherDate]? = nil
}

struct WeatherDate: Codable {
    var date: String? = nil
    var value: Double?  = nil
}
